inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -19,78 +18,254 @@
+        
+        rs: list[RepeatedSchedule] = []
+        for row in rs_rows:
+            sql = ''' SELECT * FROM repeated_schedule_manifest WHERE schedule_id = ? '''
+            rsm_rows = self._select_manifest_ids_by_schedule_id(sql, row[0])            
+            manifests = self._select_manifests_by_schedule_id(rsm_rows)                
+            rs.append(RepeatedSchedule(id=row[0], request_id=row[1], cron_duration=row[2], cron_minutes=row[3], 
+                                       cron_hours=row[4], cron_day_month=row[5], cron_month=row[6], cron_day_week=row[7],
+                                       manifests=manifests))
+        return rs
+        
+    def _select_manifest_ids_by_schedule_id(self, sql: str, schedule_id: int) -> list[int]:
+        # Get the manifest_ids from the join table
+        try:
+            self._cursor.execute(sql, (schedule_id,))
+        except (sqlite3.IntegrityError, sqlite3.InternalError, sqlite3.OperationalError) as e:
+            raise DispatcherException(f"Error selecting schedule manifest from database: {e}")
+        
+        rows = self._cursor.fetchall()
+        if len(rows) == 0:
+            raise DispatcherException(f"No schedule manifest found with schedule_id: {schedule_id}")
+        return rows
+    
+    def _select_manifests_by_schedule_id(self, schedule_manifest_rows: list[Any]) -> list[str]:
+        # Get the manifests using the manifest_ids from the join table
+        manifests: list[str] = []
+        for row in schedule_manifest_rows:
+            # Get the manifests using the manifest_ids from the join table
+            sql = ''' SELECT * FROM manifest WHERE id = ? '''
+            try:
+                self._cursor.execute(sql, (row[2],))
+            except (sqlite3.IntegrityError, sqlite3.InternalError, sqlite3.OperationalError) as e:
+                raise DispatcherException(f"Error selecting manifest from database: {e}")
+            
+            manifest_row = self._cursor.fetchone()
+            if not manifest_row:
+                raise DispatcherException(f"No manifest found with id: {manifest_row[2]}")
+            manifests.append(manifest_row[1])
+        return manifests  
+                 
+    def _create_tables_if_not_exist(self) -> None:
+        self._create_single_schedule_table()
+        self._create_repeated_schedule_table()
+        self._create_manifest_table()
+        self._create_single_schedule_manifest_table()
+        self._create_repeated_schedule_manifest_table()
+    
+    def _create_single_schedule_table(self) -> None:      
+        sql = ''' CREATE TABLE IF NOT EXISTS single_schedule(
+                                id INTEGER PRIMARY KEY AUTOINCREMENT,
+                                request_id TEXT NOT NULL,
+                                start_time TEXT NOT NULL,
+                                end_time TEXT); '''
+        self._conn.execute(sql)
```

Comment:
<h3>$\color{#FFA500}Improvements: Code Maintainability$</h3>

Consider adding more detailed logging, especially before performing database operations, to aid in debugging and operational monitoring.